### PR TITLE
improved algorithm

### PR DESCRIPTION
**Pull request by mak-skid**
Used bisect for the current_time function

### CODE REVIEW

File: json_asset.py [hunk 1]
```diff
@@ -1,29 +1,24 @@
...
-        if now_minute in [0, 1, 2, 3]:
-            preformatted = datetime.now(JST) + timedelta(minutes=-5)
-            current_time = str(preformatted.strftime('%Y%m%d%H')) + str(frequency[-1]-1).zfill(2)
-            break
-        elif now_minute in [58, 59]:
-            current_time = str(datetime.now(JST).strftime('%Y%m%d%H')) + str(frequency[-1]-1).zfill(2)
-            break
-        elif now_minute > minute: #3:12
-            continue
+    def helper(now_minute=datetime.now(JST).minute):
```
Comment (IPINGCHOU) [inline json_asset.py]: Do we need a nested function? Or we could just write it as a part of the 
```python
def current_time()
```

File: json_asset.py [hunk 2]
```diff
@@ -1,29 +1,24 @@
...
+        if i == 0:
+            return -1, freq[-1]-1
         else:
-            if now_minute > frequency[i-1]:
-                current_time = str(datetime.now(JST).strftime('%Y%m%d%H')) + str(frequency[i-1]-1).zfill(2)
-                break
+            return 0, freq[i-1]-1
+
+    preformatted = datetime.now(JST) + timedelta(hours=helper()[0])
+    current_time = str(preformatted.strftime('%Y%m%d%H')) + str(helper()[1]).zfill(2)
```
Comment (IPINGCHOU) [inline json_asset.py]: The `helper()` has been called twice with the same output.
You can save the result temporarily somewhere to save the computation.

```python
fetched_time = helper()
preformatted = datetime.now(JST) + timedelta(hours = fetched_time[0])
current_time = str(preformatted.strftime('%Y%m%d%H')) + str(fetched_time[1]).zfill(2)
```

**Review by IPINGCHOU — COMMENTED** [review]
Comments left ;)

File: json_asset.py (hunk 1 above)
Comment (mak-skid) [reply json_asset.py]: True, I didn't realise I could have just assigned the value to some property i.e. fetched_time 😂

**Review by IPINGCHOU — APPROVED** [review]
ＬＧＴＭ

**Pull request merged**